onnxruntime/core/common/profiler.cc

```diff
@@ -2,11 +2,148 @@
 // Licensed under the MIT License.
 
 #include "profiler.h"
+#include <cmath>
+
+#ifdef USE_CUDA
+#include <cupti.h>
+#endif
 
 namespace onnxruntime {
 namespace profiling {
 using namespace std::chrono;
 
+class DeviceProfiler {
```

Comment:
>DeviceProfiler [](start = 6, length = 14)

should add some UT for this with CUDA build.

Reply:
Added.